src/app/core/status/status.component.ts

```diff
@@ -7,9 +11,56 @@ import { Component, OnInit } from '@angular/core';
 })
 export class StatusComponent implements OnInit {
 
-  constructor() { }
+  private peerListCount: number = 0;
+  private _subPeerList: Subscription;
+
+  private encryptionStatus: string = '_off';
+
+  constructor(private  _peerService: PeerService, private _rpc: RPCService) { }
 
   ngOnInit() {
+    this._subPeerList = this._peerService.getPeerList()
+      .subscribe(
+        peerList => {
+          this.peerListCount = peerList.length;
+        },
+        error => console.log('StatusComponent subscription error:' + error));
+
+    this._rpc.register(this, 'getwalletinfo', null, this.rpc_walletEncryptionStatus, 'both'); ;
+  }
+
+  getIconNumber(): string {
+    if (this.peerListCount <= 0) {
+      return '0';
+    } else if (this.peerListCount === 1) {
+      return '1';
+    } else if (this.peerListCount <= 3) {
+      return '2';
+    } else if (this.peerListCount === 4) {
+      return '3';
+    } else if (this.peerListCount <= 6) {
+      return '4';
+    } else if (this.peerListCount === 7) {
+      return '5';
+    } else if (this.peerListCount >= 8) {
+      return '6';
+    }
+  }
+
+  getIconLock(): string {
```

Comment:
why are we using a get function to retrieve the variable? just use the variable.